x-pack/legacy/plugins/siem/server/lib/detection_engine/routes/rules/utils.ts

```diff
@@ -118,29 +136,53 @@ export const transformAlertsToRules = (
   return alerts.map(alert => transformAlertToRule(alert));
 };
 
-export const transformFindAlertsOrError = (findResults: { data: unknown[] }): unknown | Boom => {
-  if (isAlertTypes(findResults.data)) {
-    findResults.data = findResults.data.map(alert => transformAlertToRule(alert));
+export const transformFindAlertsOrError = (
+  findResults: { data: unknown[] },
+  ruleStatuses?: unknown[]
+): unknown | Boom => {
+  if (!ruleStatuses && isAlertTypes(findResults.data)) {
+    findResults.data = findResults.data.map(
+      alert => transformAlertToRule(alert) // fix type
```

Comment:
I would use `TODO: fix type` as I typically grep through the code using TODO to find what's left over.

Reply:
Yup. Removing the comment now since adding the right saved objects types.